Updated to new tooltips

barklund opened the pull request:
## Context

This updates the icons in the page and carousel menu to the new tooltips.

Note 1: Updating the tooltip for the keyboard shortcut dialog toggle is implemented in #6403.

## User-facing changes

Hover the icons in the page action menu (remove page, duplicate page, add page, undo, redo, play/stop animation) or in the carousel menu (open grid view, toggle meta-boxes) and see the difference from the old dark tooltips to the new bright tooltips.

## Testing Instructions

### QA

This PR can be tested by following these steps:

1. Hover the icons in the page action menu (remove page, duplicate page, add page, undo, redo, play/stop animation) or in the carousel menu (open grid view, toggle meta-boxes)
2. Observe the new bright tooltips

### UAT

<!-- ignore-task-list-start -->
- [x] UAT should use the same steps as above.
<!-- ignore-task-list-end -->

## Checklist

<!-- Check these after PR creation -->

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6377

Commit:
Added new arrows using design system

Commit:
Using existing deepmerge lib, but not actively adding to package.json

This is only temporary until we come up with the proper solution in #6209

Commit:
Added new arrow buttons and spacing to carousel

Commit:
Changed buttons to circles because I think that is the intent?

Commit:
Updated to new icon buttons in the page menu

They do not look very good though. Will have to ask about that.

Commit:
Added page menu icons

Had to change some things around the page menu layout as well, but some of those overlaps with zoom PR i #4022, so will have some conflicts there.

Commit:
Generalized common props

Commit:
Merge branch 'main' into add/6204-workspace-icons

Commit:
Removed unused flag

Commit:
Added new button pixels constants

Commit:
Employed new button pixel constants

Commit:
Added one more case of button pixel usage

Commit:
Changed all buttons to square (never circles!)

Commit:
Removed irrelevant css property

Commit:
Much better grid structure!

Commit:
Moved dashboard specific icons out of design system

Commit:
Replaced all icons with new ones

This has many consequences as the new icons are differently boxed compared to the old ones. I will try to fix in subsequent commits.

Commit:
Fixed icon styling

Commit:
Added small icons, fixed dropdown chevron

Commit:
Fixed cross icon in snackbar

Commit:
Changed all icon colors to dynamic and added storybook demo

Commit:
Fixed inspector icon usage and storybook

Commit:
Fixed plus icon in color picker

Commit:
Fixed xs chevron icons after alignment

Commit:
Fixed plus icons in color preset empty panel

Commit:
Fixed icons in preset panel header

Commit:
Swapped two icons

Commit:
Fixed design in help center including new icons

Commit:
Reversed help center chevron direction

Commit:
Renamed icon

Commit:
Merge branch 'main' into task/6299-ds-icons

Commit:
Updated instructions

BrittanyIRL committed:
Made some icon adjustments

Make checkmark bigger in design system (used on dashboard/settings and telemetry banner), make toggle checkmark bigger (only exists in storybook), tweak prepublish alert icon that shows up on publisher or update button to be better size, move space for help center toggle to be padding instead of a div. (#6319)

Commit:
Edited icon instructions

Commit:
Merge branch 'task/6299-ds-icons' of github.com:google/web-stories-wp into task/6299-ds-icons

Commit:
Merge branch 'task/6299-ds-icons' into add/6204-workspace-icons

# Conflicts:
#	assets/src/design-system/components/button/index.js
#	assets/src/edit-story/components/canvas/pagemenu/index.js
#	assets/src/edit-story/components/canvas/pagenav/index.js
#	assets/src/edit-story/theme.js

Commit:
Fixed page menu sizing and structure

Commit:
Refactored page menu

Commit:
Fixed page count spacing

Commit:
Updated page count to reflect recent UX review

Commit:
Fixed some icon colors

Commit:
Fixed some icon colors

Commit:
Merge branch 'task/6299-ds-icons' into add/6204-workspace-icons

Commit:
Fixed icon size in storybook

Commit:
Merge branch 'main' into task/6299-ds-icons

Commit:
Renamed some icons

Commit:
Fixed position and css for ds toggle checkmark

Commit:
Scaled page layouts icon back up

Commit:
Updated styling of + in color picker text

Commit:
Updated help center toggle button layout

Commit:
Exit debug

Commit:
Merge branch 'task/6299-ds-icons' into add/6204-workspace-icons

Commit:
Used existed typography

Commit:
Merge branch 'main' into add/6204-workspace-icons

# Conflicts:
#	assets/src/design-system/components/button/index.js

Commit:
Merge branch 'main' into add/6204-workspace-icons

# Conflicts:
#	assets/src/design-system/components/button/constants.js
#	assets/src/edit-story/components/canvas/pagemenu/index.js
#	assets/src/edit-story/components/carousel/carouselMenu.js

Commit:
Updated to new tooltips

Commit:
Added missing tail to metabox toggle

Commit:
Merge branch 'main' into redesign/6377-workspace-tooltips

# Conflicts:
#	assets/src/edit-story/components/canvas/pagemenu/pageMenuButton.js
#	assets/src/edit-story/components/carousel/carouselMenu.js

miina commented on assets/src/edit-story/components/canvas/pagemenu/pageMenuButton.js:
```diff
@@ -27,12 +27,12 @@ import {
 function PageMenuButton({ children, title, shortcut, ...rest }) {
   return (
-    <WithTooltip title={title} shortcut={shortcut}>
+    <Tooltip title={title} shortcut={shortcut} hasTail>
```
Hm, the tail of these buttons is separated with a border for some reason, whereas it doesn't happen elsewhere -- perhaps related to the placement of it and an issue with the tooltip itself?
![Screenshot 2021-02-17 at 13 47 54](https://user-images.githubusercontent.com/3294597/108200299-8f2d4b80-711e-11eb-8a27-b145cdf6e831.png)

swissspidy replied:
Probably inheriting the border from a parent? Could be solved by adding an explicit `border: none;` to `TooltipContainer`

barklund replied:
I kind of hoped it was only on my machine, but I guess not.

It's not a border, it's a gap between the elements.

And I don't know why it shows here, but not in the storybook. If this offset is lowered to 7px, there's no gap. But that's a calculated value based on the height of the tail, so it doesn't really make sense:

| `top: 8px` | `top: 7px` |
|-|-|
| ![Screen Shot 2021-02-17 at 10 13 31](https://user-images.githubusercontent.com/637548/108224668-27203a80-7109-11eb-9e49-c1be7a211797.png) | ![Screen Shot 2021-02-17 at 10 13 42](https://user-images.githubusercontent.com/637548/108224680-29829480-7109-11eb-9705-fadbde0f860b.png) |

BrittanyIRL replied:
well this is odd! I will play with this a bit and find a solution. Do you want me to do this separately in a new ticket or just post up the fix here to add in this branch?

BrittanyIRL replied:
fix here to merge into your branch: https://github.com/google/web-stories-wp/pull/6416

barklund replied:
@BrittanyIRL, thank you so much for the quick action here! Merged!

BrittanyIRL replied:
Glad I could fix it real fast! weird that it's different in the app - sorry about that!

BrittanyIRL committed:
Fixed tooltip tails (#6416)

top and bottom tail tooltips were off by 1 in app usage but not storybook - the left and right tails were already updated with - 1px but storybook made it seem that it was only an issue with l/r - this is because the SVG has 1px space below it. Also updating span to use forwardAs so we get font back.

miina reviewed (approved):
LGTM!
